tests/test_co2eq_parameters.py

```diff
@@ -265,6 +265,16 @@ def check_emission_factors(cls, callback):
             for mode, factors in modes_to_factors.items():
                 callback(mode, factors, zone)
 
+    def test_emission_factor_values_are_in_range(self):
+        """Checks all emission factors are in the allowed range."""
+
+        def callback(_mode, factors, _zone):
+            for factor in factors:
+                self.assertGreaterEqual(factor["value"], 0.0)
+                self.assertLessEqual(factor["value"], 2000.0)
```

Comment:
I wanted to check what the actual lows and highs are here, so I iterated though all values and checked mins and maxes per mode.

Some of my notes about interesting cases and things of note:

 - Battery and hydro charge is always supposed to be zero (with note: emissions are counted at discharge).
 - Values for low-carbon sources are always 0 in defaults.yaml -- should this be reconsidered?
 - Values for biomass are sometimes 0 in some zones -- but are a whopping 1000+ in US-FLA. Unless this is a mistake, I'd be really curious about why this is. The source for this is eGrid 2020 
 - https://drive.google.com/file/d/1_FAAuTtIdMuORZKsqHnd9vMNHlQgYarh/view?usp=sharing which is an ipynb file.
 - Battery discharge is sometimes greater than 0 but less than 4 in some zones.

What do you think of this version of the test?

Reply:
I think the test looks good, unless I am mistaken the 0 emissions are only for direct emissions and not lifecycle emissions. Lifecycle emissions is what is shown on the app but we also have direct emission factors for other use cases.

If it is possible to split out the lifecycle emissions and direct emissions into their own value ranges that would be even better but I think this is a good starting point and if you want to merge it as is I'm up for that. We can always improve it further down the line.

Reply:
I checked what the current ranges are for lifecycle emissions, and indeed they are non-zero (although "battery charge" and "hydro charge" are always zero). And for direct emissions, hydro, solar, wind, biomass and nuclear are all zero, and geothermal is usually zero but not always for some reason.

I could make changes in this PR, or commit this as-is and follow up to add different ranges for direct and lifecycle; either way.

Reply:
@pierresegonne what do you think?

Reply:
Hmm I think that we can take the split between direct and lifecycle in a follow-up PR. 
It could make the test more relevant as we know for example that direct emission factors for all carbon-free sources should be 0.
But the test as is already has value so let's merge it if Viktor is happy with it :)

Reply:
> Values for biomass are sometimes 0 in some zones -- but are a whopping 1000+ in US-FLA. Unless this is a mistake, I'd be really curious about why this is. The source for this is eGrid 2020

Biomass emission factors are quite complex, as there is a lot of variability in the actual fuel types burned labeled as "biomass" it can go from car tires to wood pellets to agricultural waste products.

So depending on what is burned, different rules for how to report the emissions can be applied, resulting in these large variances